File: apps/mybookkeeper/frontend/src/shared/types/tax/tax-completeness.ts
export interface FormCompleteness {
  form_name: string;
  instance_label: string | null;
  filled_fields: string[];
  missing_fields: string[];
  total_expected: number;
  total_filled: number;
  highlights: string[];
}

export interface TaxCompletenessResponse {
  tax_year: number;
  forms: FormCompleteness[];
  summary: string;
}

File: apps/mybookkeeper/backend/app/schemas/tax/tax_completeness.py
from pydantic import BaseModel


class FormCompleteness(BaseModel):
    form_name: str
    instance_label: str | None
    filled_fields: list[str]
    missing_fields: list[str]
    total_expected: int
    total_filled: int
    highlights: list[str]


class TaxCompletenessResponse(BaseModel):
    tax_year: int
    forms: list[FormCompleteness]
    summary: str
